src/modules/enrollment/components/staffAssignment/EditStaffAssignmentDialog.tsx

```diff
@@ -0,0 +1,138 @@
+import UnassignStaffButton from '@/modules/enrollment/components/staffAssignment/UnassignStaffButton';
+import EnrollmentDateRangeWithStatus from '@/modules/hmis/components/EnrollmentDateRangeWithStatus';
+import {
+  clientBriefName,
+  parseAndFormatDate,
+  parseAndFormatDateRange,
+} from '@/modules/hmis/hmisUtil';
+import {
+  HouseholdWithStaffAssignmentsFragment,
+  RelationshipToHoH,
+  StaffAssignmentDetailsFragment,
+} from '@/types/gqlTypes';
+
+export const STAFF_ASSIGNMENT_COLUMNS: Record<
+  string,
+  ColumnDef<StaffAssignmentDetailsFragment>
+> = {
+  staffName: {
+    header: 'Staff Name',
+    render: (assignment) => assignment.user.name,
+  },
+  role: {
+    header: 'Role',
+    render: (assignment) => assignment.staffAssignmentType,
+  },
+  assignmentDate: {
+    header: 'Assignment Date',
+    render: (assignment) => parseAndFormatDate(assignment.assignedAt),
+  },
+  assignmentDateRange: {
+    header: 'Assignment Dates',
+    render: (assignment) =>
+      parseAndFormatDateRange(assignment.assignedAt, assignment.unassignedAt),
+  },
+  action: {
+    header: 'Action',
+    render: (assignment) => (
+      <UnassignStaffButton staffAssignmentId={assignment.id} />
+    ),
+  },
+};
+
+interface EditStaffAssignmentDialogProps {
+  household: HouseholdWithStaffAssignmentsFragment;
+  open: boolean;
+  onClose: VoidFunction;
+}
+
+const EditStaffAssignmentDialog: React.FC<EditStaffAssignmentDialogProps> = ({
+  household,
+  open,
+  onClose,
+}) => {
+  const isTiny = useIsMobile('sm');
+
+  const headOfHousehold = useMemo(() => {
+    return household.householdClients.find(
+      (c) => c.relationshipToHoH === RelationshipToHoH.SelfHeadOfHousehold
+    );
+  }, [household.householdClients]);
+
+  const { enrollment } = useEnrollmentDashboardContext();
+
+  if (!enrollment) return;
+  if (!headOfHousehold) return;
+
+  return (
+    <CommonDialog fullWidth open={open} onClose={onClose}>
+      <DialogTitle>Manage Staff Assignments</DialogTitle>
+      <DialogContent>
+        <Stack gap={2} sx={{ mt: 2 }}>
+          <Stack direction={isTiny ? 'column' : 'row'} gap={isTiny ? 2 : 4}>
+            <CommonLabeledTextBlock title='Head of Household'>
+              {clientBriefName(headOfHousehold.client)}
+            </CommonLabeledTextBlock>
+            <CommonLabeledTextBlock title='Project'>
+              {enrollment.project.projectName}
+            </CommonLabeledTextBlock>
+            <CommonLabeledTextBlock title='Enrollment Period'>
+              <EnrollmentDateRangeWithStatus enrollment={enrollment} />
+            </CommonLabeledTextBlock>
+          </Stack>
+          <NewStaffAssignmentForm
+            householdId={household.id}
+            projectId={enrollment.project.id}
+          />
+          <Typography
+            id='current-staff-assignments'
+            sx={{ mt: 1 }}
+            variant='h5'
```

Comment:
To me, this title feels like it's too large but perhaps I'm just reacting to having so much in one modal. Also I don't know that I can suggest another typographic variant that would be more suitable so maybe we can leave it for now.

Reply:
Ok sounds good. Thanks very much @ttoomey for your thorough review and all your comments, I really appreciate it!